framework/db/src/com/cloud/utils/db/Filter.java

```diff
@@ -89,7 +89,7 @@ public void addOrderBy(Class<?> clazz, String field, boolean ascending) {
         if (_orderBy == null) {
             _orderBy = order.insert(0, " ORDER BY ").toString();
         } else {
-            _orderBy = order.insert(0, _orderBy).toString();
+            _orderBy = order.insert(0, _orderBy + ", ").toString();
```

Comment:
LGTM, but since this is a core change can you write a small unit test for this method? For the list template api, the temp_zone_pair should give a unique id (template_id + "_" + zone_id) so sorting on this should give us deterministic results.